File: test/osroom-dev/apps/admin_pages/static/js/osr_page_js/theme-setting/themes.js

    var vue = new Vue({
      el: '#app',
      delimiters:['{[', ']}'],
      data:{
          themes:[],
          img_w_h:"?w=0&h=720",
      },
      filters: {
            formatDate: function (time) {
              return irrformatDate(time, "yyyy-MM-dd hh:mm");
            }
      }
    })

    // 加载完页面执行
    $(document).ready(function(){

        get_themes();

    })

    function get_themes(){
       var d = {}
       var result = osrHttp("GET","/api/admin/theme",d, args={not_prompt:true});
       result.then(function (r) {
            vue.themes = r.data.themes;
       });

    }

    function upload_theme(){
        var formData = new FormData();
        var name = $("#upfile").attr("name");
        formData.append("upfile", $("#upfile")[0].files[0]);
        var result = osrHttpUpload("POST","/api/admin/theme", formData);
        result.then(function (r) {
            if(r.data.msg_type=="s"){
                get_themes();
            }
        });
    }

    function restore(name){
        var d = {
            theme_name:name,
            restore_deled:1
        }
        var result = osrHttp("PUT","/api/admin/theme", d);
        result.then(function (r) {
            if(r.data.msg_type=="s"){
                get_themes();
            }
        });
    }


    function delete_theme(name){
        var d = {
            theme_name:name
        }
        var result = osrHttp("DELETE","/api/admin/theme", d);
        result.then(function (r) {
            if(r.data.msg_type=="s"){
                get_themes();
            }
        });
    }


    function switch_theme(name){
        var d = {
            theme_name:name
        }
        var result = osrHttp("PUT","/api/admin/theme", d);
        result.then(function (r) {
            if(r.data.msg_type=="s"){
                get_themes();
            }
        });
    }

    $(function() {
        $('#upfile').on('change', function() {
          var fileNames = '';
          $.each(this.files, function() {
            fileNames += '<span class="badge">' + this.name + '</span> ';
          });
          $('#file-list').html(fileNames);
        });
    });

File: test/osroom-dev/apps/modules/theme_setting/apis/themes.py
#!/usr/bin/env python
# -*-coding:utf-8-*-
# @Time : 2017/11/1 ~ 2019/9/1
# @Author : Allen Woo
from flask import request
from apps.core.flask.login_manager import osr_login_required
from apps.configs.sys_config import METHOD_WARNING
from apps.core.blueprint import api
from apps.core.flask.permission import permission_required
from apps.core.flask.response import response_format
from apps.modules.theme_setting.process.nav_setting import get_navs, nav_setting, del_navs
from apps.modules.theme_setting.process.themes import get_themes, switch_theme, upload_theme, delete_theme, \
    get_theme_readme, restore_del_default_settings, get_theme_names


@api.route('/admin/theme-names', methods=['GET'])
@osr_login_required
@permission_required()
def api_get_theme_names():
    """
    GET:
        获取当前所有主题名称

    :return:
    """

    if request.c_method == "GET":
        data = get_theme_names()
    else:
        data = {"msg_type": "w", "msg": METHOD_WARNING, "custom_status": 405}
    return response_format(data)


@api.route('/admin/theme', methods=['GET', 'POST', "PUT", "DELETE"])
@osr_login_required
@permission_required()
def api_get_themes():
    """
    主题管理
    GET:
        获取当前所有主题
    POST:
        主题安装
        upfile:<file>, 上传的主题文件
    PUT:
        切换主题
        theme_name:<str>, 主题名称

        恢复已删除的默认设置
        restore_deled:<0 or 1>
        theme_name:<str>, 主题名称
    DELETE:
        删除主题
        theme_name:<str>, 主题名称
    :return:
    """

    if request.c_method == "GET":
        if request.argget.all('name'):
            data = get_theme_readme()
        else:
            data = get_themes()

    elif request.c_method == "POST":
        data = upload_theme()

    elif request.c_method == "PUT":
        if request.argget.all('restore_deled'):
            data = restore_del_default_settings()
        else:
            data = switch_theme()
    elif request.c_method == "DELETE":
        data = delete_theme()
    else:
        data = {"msg_type": "w", "msg": METHOD_WARNING, "custom_status": 405}
    return response_format(data)


@api.route('/admin/theme/nav', methods=['GET', 'POST', "PUT", "DELETE"])
@osr_login_required
@permission_required()
def api_theme_nav():
    """
    主题导航栏设置
    GET:
        获取导航栏
        theme_name:<str>, 主题名称
        language:<str>, 语言
    PUT:
        添加或更新导航栏
        theme_name:<str>, 主题名称
        language:<str>, 语言
        display_name:<str>, 显示名称
        json_data:<json>, 具体内容, 根据主题需要的格式设置
    DELETE:
        删除导航栏
        ids:<array>, [<str id>, <str id>,...]
    :return:
    """

    if request.c_method == "GET":
        data = get_navs()

    elif request.c_method in ["POST", "PUT"]:
        data = nav_setting()
    elif request.c_method == "DELETE":
        data = del_navs()
    else:
        data = {"msg_type": "w", "msg": METHOD_WARNING, "custom_status": 405}

    return response_format(data)
